hugegraph-api/src/main/java/com/baidu/hugegraph/license/LicenseVerifier.java

```diff
@@ -0,0 +1,127 @@
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.license;
+
+import java.io.File;
+import java.io.InputStream;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.prefs.Preferences;
+
+import org.slf4j.Logger;
+
+import com.baidu.hugegraph.HugeException;
+import com.baidu.hugegraph.config.HugeConfig;
+import com.baidu.hugegraph.util.E;
+import com.baidu.hugegraph.util.JsonUtil;
+import com.baidu.hugegraph.util.Log;
+
+import de.schlichtherle.license.CipherParam;
+import de.schlichtherle.license.DefaultCipherParam;
+import de.schlichtherle.license.DefaultKeyStoreParam;
+import de.schlichtherle.license.DefaultLicenseParam;
+import de.schlichtherle.license.KeyStoreParam;
+import de.schlichtherle.license.LicenseContent;
+import de.schlichtherle.license.LicenseParam;
+
+public class LicenseVerifier {
+
+    private static final Logger LOG = Log.logger(LicenseVerifier.class);
+
+    private static final String LICENSE_PARAM_PATH = "/verify-license.json";
+
+    private static volatile LicenseVerifier INSTANCE = null;
+
+    private static final Duration CHECK_INTERVAL = Duration.ofMinutes(10);
+    private static volatile Instant lastCheckTime = Instant.now();
+
+    private final LicenseVerifyParam verifyParam;
+    private final LicenseVerifyManager manager;
+
+    private LicenseVerifier(HugeConfig config) {
+        E.checkNotNull(config, "config");
+        this.verifyParam = buildVerifyParam(LICENSE_PARAM_PATH);
+        LicenseParam licenseParam = this.initLicenseParam(this.verifyParam);
+        this.manager = new LicenseVerifyManager(licenseParam, config);
+    }
+
+    public static LicenseVerifier instance(HugeConfig config) {
+        if (INSTANCE == null) {
+            synchronized(LicenseVerifier.class) {
+                if (INSTANCE == null) {
+                    INSTANCE = new LicenseVerifier(config);
+                }
+            }
+        }
+        return INSTANCE;
+    }
+
+    public static void verifyIfNeeded() {
+        Instant now = Instant.now();
+        Duration interval = Duration.between(lastCheckTime, now);
+        if (!interval.minus(CHECK_INTERVAL).isNegative()) {
+            LicenseVerifier.instance(null).verify();
```

Comment:
may it cause exception at line 59 checkNotNull()?

Reply:
when pass param null, INSTANCE has been constructed